xwiki-platform-core/xwiki-platform-repository/xwiki-platform-repository-server-api/src/main/java/org/xwiki/repository/internal/RepositoryManager.java

```diff
@@ -1102,4 +1239,42 @@ private boolean updateCollection(BaseObject extensionObject, String fieldName, C
 
         return needSave;
     }
+
+    public BaseObject getExtensionVersionObject(XWikiDocument extensionDocument, String version)
+    {
+        return getExtensionVersionObject(extensionDocument, version, true);
+    }
+
+    public BaseObject getExtensionVersionObject(XWikiDocument extensionDocument, String version, boolean allowProxying)
+    {
+        if (version == null) {
+            List<BaseObject> objects =
+                    extensionDocument.getXObjects(XWikiRepositoryModel.EXTENSIONVERSION_CLASSREFERENCE);
+
+            if (objects == null || objects.isEmpty()) {
+                return null;
+            } else {
+                return objects.get(objects.size() - 1);
+            }
+        }
+
+        BaseObject extensionVersionObject =
+                extensionDocument.getObject(XWikiRepositoryModel.EXTENSIONVERSION_CLASSNAME, "version", version, false);
+
+        if (extensionVersionObject == null && allowProxying && isVersionProxyingEnabled(extensionDocument)) {
+            //no ExtensionVersionClass object so we need to create such object temporarily and delete it
+            try {
+                //FIXME - see XWIKI-14138
+                XWikiDocument extensionDocumentClone = extensionDocument.clone();
```

Comment:
The garbage collector will take care of it.

About the comment I would prefer a more explicit test explaining it's a hack and we need to refactor the code to something cleaner.